docs/feature_rgb_matrix.md

```diff
@@ -742,6 +742,27 @@ void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
 }
 ```
 
+Layer indicator with only configured keys:
+```c
+void rgb_matrix_indicators_user(void) {
+    if (get_highest_layer(layer_state) > 0) {
+        for (uint8_t row = 0; row < MATRIX_ROWS; ++row) {
+            for (uint8_t col = 0; col < MATRIX_COLS; ++col) {
+
+                uint8_t index = g_led_config.matrix_co[row][col];
+                uint8_t layer = get_highest_layer(layer_state);
+
+                if (index >= led_min && index <= led_max && index != NO_LED &&
+                keymap_key_to_keycode(layer, (keypos_t){col,row}) > KC_TRNS) {
```

Comment:
If you don't want to change this to the advanced function you should be able to change this to
```suggestion
                if (index != NO_LED && keymap_key_to_keycode(layer, (keypos_t){col,row}) > KC_TRNS) {
```
This will likely come with some performance impact, but from memory it was okay, probably want to test this change though.

Reply:
There will be less loops to iterate for `rgb_matrix_indicators_user` like this example:
```c
		for (uint8_t row = 0; row < MATRIX_ROWS; ++row) {
			for (uint8_t col = 0; col < MATRIX_COLS; ++col) {
				if (g_led_config.matrix_co[row][col] != NO_LED &&
					keymap_key_to_keycode(layer, (keypos_t){col,row}) > KC_TRNS) {
					rgb_matrix_set_color(g_led_config.matrix_co[row][col], RGB_LAYER);
				}
			}
		}
```
However this document uses `rgb_matrix_indicators_advanced_user` consistently as examples for performance reason. Is there an indicative guide (like number of LEDs on a PCB) to using one or the other?

Reply:
Conversation resolved on discord.